### PR TITLE
fix: validate SEV cert chain and improve error output

### PR DESCRIPTION
**Pull request by haraldh**
*  feat(backend/sev): add context message to `sev::cached_chain::get()?`
    
    ```
    Error: Failed to read cached certification chain from `/var/cache/amd-sev/chain`
    
    Caused by:
        Permission denied (os error 13)
    ```

* fix(sev): print user friendly attestation_bridge errors

    ```
    Error: Failed to read cached certification chain from `/var/cache/amd-sev/chain`
    
    Caused by:
        Permission denied (os error 13)
    attestation_bridge error: failed to deserialize expected certificate chain
    ```

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,6 +15,13 @@ jobs:
           target: x86_64-unknown-linux-musl
           toolchain: nightly
           override: true
+      - name: "SEV: Verify certificate chain"
+        if: ${{ matrix.backend.name == 'sev' }}
+        run: |
+          ls -al /var/cache /var/cache/amd-sev /var/cache/amd-sev/chain
+          stat /var/cache/amd-sev/chain
```
Comment (connorkuehl) [inline .github/workflows/test.yml:22]: Is this needed?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,6 +15,13 @@ jobs:
...
+          cargo install sevctl
```
Comment (connorkuehl) [inline .github/workflows/test.yml:23]: `sevctl` isn't pushed to crates.io. You'll have to install via a git revision.

```console
cargo install --git https://github.com/enarx/sevctl
```

Can use the `--rev` switch to pin a certain commit to install from.

File: tests/integration_tests.rs [hunk 3]
```diff
@@ -364,3 +364,14 @@ fn memspike() {
...
+#[test]
+#[serial]
+fn check_amd_sev_chain() {
+    if Path::new("/dev/sev").exists() {
+        assert!(
+            Path::new("/var/cache/amd-sev/chain").exists(),
+            "`/var/cache/amd-sev/chain` does not exist"
+        );
+    }
+}
```
Comment (connorkuehl) [inline tests/integration_tests.rs:377]: Use the [`cached_chain` module](https://github.com/enarx/sev/blob/master/src/util/cached_chain.rs#L64) to adhere to the Enarx search path for the cert.

**Review by connorkuehl — CHANGES_REQUESTED** [review]

**Comment by haraldh** [comment]
Ok, I completely revamped the whole thing.
`enarx-keepldr` will check for a valid certificate chain and print diagnostics, why the backend couldn't be used.

**Comment by haraldh** [comment]
and revamped again

**Comment by connorkuehl** [comment]
As an alternative to managing the thread's join handle, what do you think of just changing the unwrap behavior in `unattested_launch::launch` to a custom `unwrap_or_exit` like we had in `sevctl` for a while?

**Comment by npmccallum** [comment]
@haraldh Can you rebase this so tests work?

**Comment by haraldh** [comment]
@npmccallum done

**Review by npmccallum — APPROVED** [review]

**Review by mbestavros — APPROVED** [review]

**Review by connorkuehl — APPROVED** [review]

**Review by npmccallum — APPROVED** [review]

**Pull request merged**